crates/engine/tree/src/engine.rs

```diff
@@ -162,8 +172,9 @@ where
     type Request = BeaconEngineMessage<T>;
 
     fn on_event(&mut self, event: FromEngine<Self::Request>) {
-        // delegate to the tree
-        let _ = self.to_tree.send(event);
+        // delegate to the tree, given that we are not currently using it here
+        // we explicitly drop the future to prevent warnings
+        std::mem::drop(self.to_tree.send(event));
```

Comment:
```suggestion
        let _ = self.to_tree.send(event);
```

Reply:
oh this is the async variant?

I think this should not by async send? should be unbounded channel, probably std::sync::mpsc because we want to have the tree service spawned on a regular thread

Reply:
makes sense, using `Sender` from `std::sync::mpsc` for `to_tree` now